dao/src/test/java/org/thingsboard/server/dao/service/BaseTenantServiceTest.java

```diff
@@ -275,4 +323,374 @@ public void testSaveTenantWithIsolatedProfileInMonolithSetup() {
+        assertWidgetsBundleIsDeleted(savedTenant, savedWidgetsBundle);
+        assertEntityViewIsDeleted(savedTenant, savedDevice, savedEntityView);
+        assertAssetIsDeleted(savedTenant, savedAsset);
+        assertDeviceIsDeleted(savedTenant, savedDevice);
+        assertDeviceProfileIsDeleted(savedTenant, savedDeviceProfile);
+        assertDashboardIsDeleted(savedTenant, savedDashboard);
+        assertEdgeIsDeleted(savedTenant, savedEdge);
+        assertTenantAdminIsDeleted(savedTenant);
+        assertUserIsDeleted(savedTenant, savedUser);
+        Assert.assertNull(ruleChainService.findRuleChainById(savedTenant.getId(), savedRuleChain.getId()));
+        Assert.assertNull(apiUsageStateService.findTenantApiUsageState(savedTenant.getId()));
+        assertResourceIsDeleted(savedTenant, savedResource);
+        assertOtaPackageIsDeleted(savedTenant, savedOtaPackage);
+        Assert.assertNull(rpcService.findById(savedTenant.getId(), savedRpc.getId()));
+
+        tenantProfileService.deleteTenantProfile(TenantId.SYS_TENANT_ID, savedProfile.getId());
+    }
+
+    private void assertOtaPackageIsDeleted(Tenant savedTenant, OtaPackage savedOtaPackage) {
+        Assert.assertNull(
+                otaPackageService.findOtaPackageById(
+                        savedTenant.getId(), savedOtaPackage.getId()
+                )
+        );
+        PageLink pageLinkOta = new PageLink(1000);
+        PageData<OtaPackageInfo> pageDataOta = otaPackageService.findTenantOtaPackagesByTenantId(savedTenant.getId(), pageLinkOta);
+        Assert.assertFalse(pageDataOta.hasNext());
+        Assert.assertEquals(0, pageDataOta.getTotalElements());
+    }
+
+    private void assertResourceIsDeleted(Tenant savedTenant, TbResource savedResource) {
+        Assert.assertNull(resourceService.findResourceById(savedTenant.getId(), savedResource.getId()));
+        PageLink pageLinkResources = new PageLink(1000);
+        PageData<TbResourceInfo> tenantResources =
+                resourceService.findAllTenantResourcesByTenantId(savedTenant.getId(), pageLinkResources);
+        Assert.assertFalse(tenantResources.hasNext());
+        Assert.assertEquals(0, tenantResources.getTotalElements());
+    }
+
+    private void assertUserIsDeleted(Tenant savedTenant, User savedUser) {
+        Assert.assertNull(userService.findUserById(savedTenant.getId(), savedUser.getId()));
+        PageLink pageLinkUsers = new PageLink(1000);
+        PageData<User> users =
+                userService.findUsersByTenantId(savedTenant.getId(), pageLinkUsers);
+        Assert.assertFalse(users.hasNext());
+        Assert.assertEquals(0, users.getTotalElements());
+    }
+
+    private void assertTenantAdminIsDeleted(Tenant savedTenant) {
+        PageLink pageLinkTenantAdmins = new PageLink(1000);
+        PageData<User> tenantAdmins =
+                userService.findTenantAdmins(savedTenant.getId(), pageLinkTenantAdmins);
+        Assert.assertFalse(tenantAdmins.hasNext());
```

Comment:
You are fetching the first page with data and asking is any chance to get a next page . Please, check the hasNext method source code